File: app/src/test/java/acr/browser/lightning/utils/PreconditionsTest.java
package acr.browser.lightning.utils;

import org.junit.Test;

/**
 * Unit tests for {@link Preconditions}.
 */
public class PreconditionsTest {

    @Test(expected = RuntimeException.class)
    public void checkNonNull_Null_ThrowsException() {
        Preconditions.checkNonNull(null);
    }

    @Test
    public void checkNonNull_NonNull_Succeeds() {
        Preconditions.checkNonNull(new Object());
    }
}

File: app/src/test/java/acr/browser/lightning/utils/PreconditionsTest.kt
package acr.browser.lightning.utils

import org.junit.Test

/**
 * Unit tests for [Preconditions].
 */
class PreconditionsTest {

    @Test(expected = RuntimeException::class)
    fun `checkNonNull throws exception for null param`() = Preconditions.checkNonNull(null)

    @Test
    fun `checkNonNull succeeds for non null param`() = Preconditions.checkNonNull(Any())
}